src/main.d

```diff
@@ -166,34 +179,50 @@ int main(string[] args)
 		return EXIT_FAILURE;
 	}
 	
-	// Set the local path OneDrive root
-	if (syncDirName){
-		// The user passed in an alternate sync_dir
+	// command line parameters to override the 'config' & take precedence
+	if (skipSymlinks) {
+		// The user passed in an alternate skip_symlinks as to what was either in 'config' file or application default
+		log.vdebug("CLI override to set skip_symlinks to: true");
+		cfg.setValue("skip_symlinks", "true");
+	}
+	
+	// Set the OneDrive Local Sync Directory if was passed in via --syncdir
+	if (syncDirName) {
+		// The user passed in an alternate sync_dir as to what was either in 'config' file or application default
+		// Do not expandTilde here as we do not know if we reliably can
+		log.vdebug("CLI override to set sync_dir to: ", syncDirName);
 		cfg.setValue("sync_dir", syncDirName);
 	}
 	
+	// sync_dir environment handling to handle ~ expansion properly
 	string syncDir;
 	if ((environment.get("SHELL") == "") && (environment.get("USER") == "")){
-		// No shell or user set, so expandTilde() will fail - usually headless system running under init.d / systemd
-		// Did the user specify a 'different' sync dir by passing a value in?
-		if (syncDirName){
-			// was there a ~ in the passed in state? it will not work via init.d / systemd
-			if (canFind(cfg.getValue("sync_dir"),"~")) {
-				// A ~ was found
-				syncDir = homePath ~ "/OneDrive";
-			} else {
-				// No ~ found in passed in state, use as is
-				syncDir = cfg.getValue("sync_dir");
-			}
+		log.vdebug("sync_dir: No SHELL or USER detected");
+		// No shell or user set, so expandTilde() will fail - usually headless system running under init.d / systemd or potentially Docker
+		// Does the 'currently configured' sync_dir include a ~
+		if (canFind(cfg.getValue("sync_dir"),"~")) {
+			// A ~ was found
+			log.vdebug("sync_dir: A '~' was found in configured sync_dir, using the calculated 'homePath' to replace '~'");
+			syncDir = homePath ~ strip(cfg.getValue("sync_dir"),"~");
```

Comment:
I don't know about anyone else here, as I am quite unfamiliar with dlang, L203-206 seems fragile to me, no? See below:

```
{18-12-22 11:31}linux-desktop:/tmp nouser% mkdir -p parent/~/child1/~/child2/~/child3
{18-12-22 11:32}linux-desktop:/tmp nouser% tree parent
parent
└── ~
    └── child1
        └── ~
            └── child2
                └── ~
                    └── child3

6 directories, 0 files

```

Reply:
@lhorace 
dlang has a 'real' problem with performing 'expandTilde' function if there is no SHELL or USER environment variable. Example using '~/.config/onedrive' below

In the original code (skilion) there was lines like this:
```
configDirName = configDirName.expandTilde().absolutePath();
```
In docker or headless systems, when run from cron etc, this would be interpreted as:
```
/.config/onedrive
```
The client would fail to expand the ~ as it did not know how to expand.
where in reality '~' should have been expanded to the users home directory.

So the real problem here is trying to detect ~ in syncDir & confDir parameters, and if no SHELL or USER env set, expand this out a different way.

I agree with your example, that the current 'strip' is fragile - as in it would replace all ~ in your example string with the calculated string. In my testing I was only factoring in the input variable having '~' at the start